src/ralph/dns/views.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from django.http import HttpResponseRedirect
+
+from ralph.admin.views.extra import RalphDetailView
+from ralph.dns.dnsaas import DNSaaS
+from ralph.dns.forms import DNSRecordForm
+
+logger = logging.getLogger(__name__)
+
+
+class DNSView(RalphDetailView):
+    icon = 'chain-broken'
+    name = 'dns_edit'
+    label = 'DNS'
+    url_name = 'dns_edit'
+    template_name = 'dns/dns_edit.html'
+
+    def __init__(self, *args, **kwargs):
+        self.dnsaas = DNSaaS()
+        return super().__init__(*args, **kwargs)
+
+    def get_forms(self):
+        forms = []
+        initial = self.dnsaas.get_dns_records(
+            self.object.ipaddress_set.all().values_list(
+                'address', flat=True
+            )
+        )
+        for item in initial:
+            forms.append(DNSRecordForm(item))
+
+        forms.append(DNSRecordForm())
+        return forms
+
+    def get(self, request, *args, **kwargs):
+        kwargs['forms'] = self.get_forms()
```

Comment:
shouldn't we check if forms are already in kwargs? What about the case with errors during post? Do we need to call this API second time?